File: src/test/groovy/com/github/jengelman/gradle/plugins/shadow/docs/internal/snippets/fixture/SnippetFixture.java
package com.github.jengelman.gradle.plugins.shadow.docs.internal.snippets.fixture;

import com.github.jengelman.gradle.plugins.shadow.docs.internal.Block;

public class SnippetFixture {

  public void around(Block action) throws Exception {
    action.execute();
  }

  public String transform(String text) {
    return text;
  }

  public String pre() {
    return "";
  }

  public String post() {
    return "";
  }

  public Integer getOffset() {
    return pre().split("\n").length;
  }

}

File: src/integrationTest/kotlin/com/github/jengelman/gradle/plugins/shadow/docs/internal/snippets/fixture/SnippetFixture.kt
package com.github.jengelman.gradle.plugins.shadow.docs.internal.snippets.fixture

interface SnippetFixture {
  fun transform(text: String): String = text

  fun pre(): String = ""

  fun post(): String = ""

  val offset: Int get() = pre().lines().dropLastWhile { it.isEmpty() }.size
}
